core-common/src/test/java/org/zalando/nakadi/domain/EventOwnerHeaderTest.java

```diff
@@ -14,7 +16,9 @@
     @Test
     public void testEventOwnerHeaderSerialization() {
         final EventOwnerHeader eventOwnerHeader = new EventOwnerHeader(NAME, VALUE);
-        final ProducerRecord<String, String> record = new ProducerRecord<>("topic", "value");
+        final ProducerRecord<byte[], byte[]> record = new ProducerRecord<>(
+                "topic",
```

Comment:
How does it work here with "topic" not converted to `byte[]`?

Reply:
conversion to bytes only for key / value, that should be non default producer constructor where key is null by default

Reply:
yes, that is exactly the case 
```
    public ProducerRecord(String topic, V value) {
        this(topic, null, null, null, value, null);
    }
```

Reply:
ah, I see.  the first argument is not key, but indeed a topic, which is expected to be of type `String`, regardless of the key and value types: https://kafka.apache.org/23/javadoc/org/apache/kafka/clients/producer/ProducerRecord.html#ProducerRecord-java.lang.String-V-
```
public ProducerRecord(String topic,
                      V value)
Create a record with no key
```